controllers/reconcile_scale_zero.go

```diff
@@ -0,0 +1,101 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/rabbitmq/cluster-operator/v2/api/v1beta1"
+	"github.com/rabbitmq/cluster-operator/v2/internal/status"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const beforeZeroReplicasConfigured = "rabbitmq.com/before-zero-replicas-configured"
+
+// scaleToZero checks if the desired replicas is zero and the current replicas is not zero.
+func (r *RabbitmqClusterReconciler) scaleToZero(current, sts *appsv1.StatefulSet) bool {
+	currentReplicas := *current.Spec.Replicas
+	desiredReplicas := *sts.Spec.Replicas
+
+	if desiredReplicas == 0 && currentReplicas > 0 {
+		return true
+	}
+
+	return false
```

Comment:
Accept it at your discretion:

```suggestion
	return desiredReplicas == 0 && currentReplicas > 0
```

Same in `scaleFromZero`

Reply:
Done